Dockerfile.upstream

```diff
@@ -0,0 +1,14 @@
+FROM golang:1.21 as builder
+WORKDIR /go/src/github.com/openshift/coredns-ocp-dnsnameresolver
+COPY . .
+
+RUN make build-coredns ARGS="upstream"
+
+FROM gcr.io/distroless/base-debian12:latest
```

Comment:
Why particular reason it's debian based?

Reply:
As upstream coredns uses debian based images, I used the same here. However, I faced some issues with glibc libraries required for go 1.21 when testing the image built using `static-debian`, that's why I used `base-debian`.

Reply:
Got it, we can have a closer look at the problem with the `static-debian` image later.